Rule/index.ts

```diff
@@ -1,48 +1,48 @@
 import { selectively } from "selectively"
 import { definitions } from "./definitions"
-import * as ModelRule from "./Rule"
+import { Rule as ModelRule, type as ruleType } from "./Rule"
 import { State as RuleState } from "./State"
 
-export type Rule = ModelRule.Rule
+export type Rule = ModelRule
 
 export namespace Rule {
 	export const actions = ModelRule.actions
 	export type Action = ModelRule.Action
 	export const kinds = ModelRule.kinds
 	export type Kind = ModelRule.Kind
-	export type State = RuleState
-	export const State = RuleState
-	export namespace State {
-		export type Account = RuleState.Account
-		export namespace Account {
-			export type Transactions = RuleState.Account.Transactions
-			export type Days = RuleState.Account.Days
-		}
-		export type Authorization = RuleState.Authorization
-		export type Card = RuleState.Card
-		export namespace Card {
-			export type Statistics = RuleState.Card.Statistics
-		}
-		export type Transaction = RuleState.Transaction
-		export type Data = RuleState.Data
-		export type Partial = RuleState.Partial
-		export type Organization = RuleState.Organization
+	export import State = RuleState
+	export const type = ruleType
+	export const is = ruleType.is
+	export const flaw = ruleType.flaw
+	function resolve(rule: ModelRule, state: State, macros?: Record<string, selectively.Definition>) {
+		return selectively.resolve({ ...macros, ...definitions }, selectively.parse(rule.condition)).is(state)
+	}
+	function score(
+		rules: (Rule & ModelRule.Score)[],
+		state: State,
+		macros?: Record<string, selectively.Definition>
+	): number | undefined {
+		return rules.reduce(
+			(r: number | undefined, rule) => (resolve(rule, state, macros) ? (r ?? 100) * (rule.risk / 100) : undefined),
+			undefined
+		)
 	}
-
-	export const type = ModelRule.type
-	export const is = ModelRule.type.is
-	export const flaw = ModelRule.type.flaw
 	export function evaluate(
 		rules: Rule[],
 		state: State,
 		macros?: Record<string, selectively.Definition>
-	): Record<Action, Rule[]> {
-		const result: Record<Action, Rule[]> = { review: [], reject: [], flag: [] }
-		rules.forEach(
-			r =>
-				selectively.resolve({ ...macros, ...definitions }, selectively.parse(r.condition)).is(state) &&
-				result[r.action].push(r)
+	): Record<Action, Rule[]> & { risk?: number } {
```

Comment:
I think this will be easier to handle if you make the return type { action: Record<Action, Rule[]>, risk?: number }}.